addons/account/models/account_journal.py

```diff
@@ -523,7 +523,7 @@ def create(self, vals):
 
         journal = super(AccountJournal, self.with_context(mail_create_nolog=True)).create(vals)
 
-        if 'alias_name' in vals:
+        if journal.type in ('sale', 'purchase'):
```

Comment:
Are you sure it's not breaking anything if `'alias_name'` is not in vals ?

Reply:
https://github.com/odoo/odoo/blob/5e684773fca39bc15f99d7c83889818b351b424a/addons/account/models/account_journal.py#L371-L373

then

https://github.com/odoo/odoo/blob/5e684773fca39bc15f99d7c83889818b351b424a/addons/account/models/account_journal.py#L330-L334